robolectric-shadows/shadows-core/src/main/resources/org/robolectric/fakes/RenderService.java.vm

```diff
@@ -0,0 +1,313 @@
+   */
+  public RenderService setDecorations(boolean show) {
+    showDecorations = show;
+    return this;
+  }
+
+  public RenderService setAppInfo(String label, String icon) {
+    appLabel = label;
+    appIconName = icon;
+    return this;
+  }
+
+  public RenderService setMinSdkVersion(int min) {
+    minSdkVersion = min;
+    return this;
+  }
+
+  public RenderService setTargetSdkVersion(int target) {
+    targetSdkVersion = target;
+    return this;
+  }
+
+  public RenderService setExactDeviceDpi(float xdpi, float ydpi) {
+    Xdpi = xdpi;
+    Ydpi = ydpi;
+    return this;
+  }
+
+  public RenderService setImageFactory(IImageFactory factory) {
+    imageFactory = factory;
+    return this;
+  }
+
+  /** Initializes any remaining optional fields after all setters have been called */
+  private void finishConfiguration() {
+    if (logger == null) {
+      // Silent logging
+      logger = new LayoutLog();
+    }
+  }
+
+  public SessionParams getParamsFromFile(String layoutName) throws FileNotFoundException, XmlPullParserException {
+    finishConfiguration();
+    if (resourceResolver == null) {
+      // Abort the rendering if the resources are not found.
+      return null;
+    }
+    // find the layout to run
+    ResourceValue value = resourceResolver.getProjectResource(ResourceType.LAYOUT, layoutName);
+    if (value == null || value.getValue() == null) {
+      throw new IllegalArgumentException("layout does not exist");
+    }
+    File layoutFile = new File(value.getValue());
+    ILayoutPullParser parser = null;
+    parser = new XmlParser();
+    parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, true);
+    parser.setInput(new FileInputStream(layoutFile), "UTF-8"); // $NON-NLS-1$
+    figureSomeValuesOut();
+    HardwareConfig hardwareConfigconfig = new HardwareConfig(config.getScreenWidthQualifier().getValue(),
+        config.getScreenHeightQualifier().getValue(), config.getDensityQualifier().getValue(),
+        Xdpi, Ydpi, config.getScreenSizeQualifier().getValue(),
+        config.getScreenOrientationQualifier().getValue(), false);
+    SessionParams params = new SessionParams(parser, renderingMode, this/* projectKey*/, hardwareConfigconfig,
+        resourceResolver, projectCallback, minSdkVersion, targetSdkVersion, logger);
+    params.setExtendedViewInfoMode(true);
+    if (!showDecorations) {
+      params.setForceNoDecor();
+    } else {
+      if (appLabel == null) {
+        appLabel = "Random App";
+      }
+      params.setAppLabel(appLabel);
+      params.setAppIcon(appIconName); // ok to be null
+    }
+    if (overrideBgColor != null) {
+      params.setOverrideBgColor(overrideBgColor.intValue());
+    }
+    // set the Image Overlay as the image factory.
+    params.setImageFactory(imageFactory);
+    return params;
+  }
+
+  /**
+   * Renders the model and returns the result as a {@link RenderSession}.
+   * @return the {@link RenderSession} resulting from rendering the current model
```

Comment:
Javadoc consistently, if you don't document the parameters then remove them from the docs